yarn-project/pxe/src/pxe_service/pxe_service.ts

```diff
@@ -526,15 +526,19 @@ export class PXEService implements PXE {
     msgSender: AztecAddress | undefined = undefined,
     skipTxValidation: boolean = false,
     scopes?: AztecAddress[],
-  ): Promise<SimulatedTx> {
+  ): Promise<TxSimulationResult> {
     return await this.jobQueue.put(async () => {
-      const simulatedTx = await this.#simulateAndProve(txRequest, this.fakeProofCreator, msgSender, scopes);
+      const privateExecutionResult = await this.#simulatePrivate(txRequest, msgSender, scopes);
+      const publicInputs = await this.#simulateKernels(txRequest, privateExecutionResult);
+      const privateSimulationResult = new PrivateSimulationResult(privateExecutionResult, publicInputs);
```

Comment:
According to @sirasistant, private functions are more "executed" than "simulated, since we do witness generation. I agree with him, solely based on the fact that renaming `#simulatePrivate` is a lot less work than renaming `PrivateExecutionResult` 🤣

Reply:
I see that making sense for someone who knows what the function does, but don't you think it's slightly strange that `executePrivate` doesn't actually execute a private function?

What if the names were a bit longer? E.g. `simulatePrivateFunction`, `simulatePrivateTransaction` (or kernels - not sure if we need to mention them here tbh), `provePrivateTransaction` etc

Reply:
Well...it does execute it (we even get return values back!)